### PR TITLE
Add option runSuiteSuffix to 'run suite' command

### PR DESCRIPTION
**Pull request by poohnix**
Problem:
There might be times when you don't want to run all of your test suites but only specific suites, say for checking units and integration but not acceptance. This can be done via the command line switch "--testsuite <name,...>" of PHPUnit.

Solution:
Add a new option to specify which suites to run with the "run suite" command or run all of them if the option "runSuites" is empty (the default).

Remark:
This patch obviously had to include my previous one for fixing the extension tests.

### CODE REVIEW

**Comment by calebporzio** [comment]
Really nice PR. Mind explaining the use case a bit more to me?

If the "run-suite" command currently runs ALL the test suites, then adding this config option would make it possible to only run SOME suites on "run-suite".

It seems like the ideal feature would be automatic suite detection and providing commands to run each suite. Unless I'm missing a strong use case for excluding acceptance tests or some kind of suite from an "run-suite" run.

Does my question make sense? I'm just trying to get a better feel for the use cases of this feature.

**Comment by poohnix** [comment]
Hi Caleb,

yeah, sorry for my very brief description in this PR, I should have made the use case clearer in the first place. So for a fresh start I'll try again. ;-)

TL;DR: It is probably rather a "problem" with [Codeception](https://codeception.com/) as it configures three/four standard suites directly from the beginning which can be time consuming if you work only on a certain aspect / within a certain suite "domain".

First of all thanks for this amazing VS-Code extension! Until I stumbled upon it by pure fortune I had no idea I needed it that badly! Within the last two days that I'm using it, it has already become a very valued asset to me - so much that I wanted to give something back as I had to change it slightly to fit my use case(s).

So back to the discussion. I'm currently developing a bigger WordPress plugin for a client. I use OOP and TDD to develop it which does not seem to be too common in the WordPress world. Anyhow I use [Codeception](https://codeception.com/) for this which is build upon PHPUnit for it's unit/wpunit testing support but adds scaffolding and functional- and acceptance testing support to it.

Because of this there exist four different test suites in my project - unit (for "normal" unit testing), wpunit (for WordPress unit testing), functional and acceptance. I guess this is rather different from pure PHPUnit testing so the use case is probably intertwined with using Codeception.

Anyway my use of your extension is this:
- I let Codeception scaffold a new wpunit test to test and develop a new class for the plugin.
_The classes are closely connected via dependency injection so I only need unit (wpunit) testing at the moment._
- I write a first test case to instantiate the new class - which fails as there is no such class.
- I scaffold the class and run the test again - which passes.
_The above is done via the extensions "run" command to run a single method._
- From then on I write further tests for new functionality, they fail, I implement the new method/functionality, "run" the single test method in the extension again, and it (hopefully ;-) passes.
- Every three to five test cases I let the extension "run" the whole class to ensure I haven't broken any previous test cases.
- Every ten to fifteen test cases I let the extension "run suite" to ensure I have not broken any interweaving functionality. <- **And this is my use case for this PR.**

Running all four test suites is taking much more time than only running the unit/wpunit suite(s) and is unnecessary in my case as the functions I'm currently working on are all rather isolated in the close scope of the dependency injection.

I hope this makes a bit more sense now, sorry for getting that lengthy. If not feel free to get back to me again and thanks for your quick reaction to my PR - I'm not used to this over here on Github. ;-)

Best regards,
Hank

**Comment by poohnix** [comment]
Sorry, forgot to answer your other questions:

> If the "run-suite" command currently runs ALL the test suites, then adding this config option would make it possible to only run SOME suites on "run-suite".

Yes, exactly. With this config option you can chose if you want to run all suites by just leaving it empty (which is the default for this option and the behaviour without this PR) or if you define a comma separated list of suites only these suites are run by the extensions "run suites" command.

> It seems like the ideal feature would be automatic suite detection and providing commands to run each suite. Unless I'm missing a strong use case for excluding acceptance tests or some kind of suite from an "run-suite" run.

Yes, the use case is just saving some time if you are working only on one aspect of the project like developing new rather isolated functionality (see my comment above). I agree that it would be a cool feature to have several commands to only run one suite at a time or all of them. If this PR is of any interest to you I can have a look into that as well.

> Does my question make sense? I'm just trying to get a better feel for the use cases of this feature.

Totally makes sense. I hope I explained everything a bit better now?

**Comment by poohnix** [comment]
Hello Caleb,

I have revised this PR to broaden the use cases of this new option. It now let's you add a command suffix to only the 'run suite' command. This is useful if you want to specify '--testsuite unit' (my previous use case) but it also let's you specify things like '--coverage' that really only make sense to be run against the whole suite and not on single files or even methods (like commandSuffix would).

I hope this broader range of use cases makes more sense to you?

Cheers,
Hank

**Comment by poohnix** [comment]
If this PR is of interest to you please have a look at PR #90 which is a streamlined version of the above changes and should be way easier to comprehend.
